sections/shared/Layout/SideNav/DesktopSideNav.tsx

```diff
@@ -27,24 +23,31 @@ import CRatioBarStats from 'sections/shared/Layout/Stats/CRatioBarStats';
 
 import SideNav from './SideNav';
 import SubMenu from './DesktopSubMenu';
+import useSynthetixQueries from '@synthetixio/queries';
+import { wei } from '@synthetixio/wei';
 
 const DesktopSideNav: FC = () => {
+	const networkId = useRecoilValue(networkState)!.id;
+
+	const walletAddress = useRecoilValue(walletAddressState);
+
+	const { useSNX24hrPricesQuery, useSynthsBalancesQuery } = useSynthetixQueries();
+
 	const SNX24hrPricesQuery = useSNX24hrPricesQuery();
-	const cryptoBalances = useCryptoBalances();
-	const synthsBalancesQuery = useSynthsBalancesQuery();
+	const cryptoBalances = useCryptoBalances(walletAddress);
+	const synthsBalancesQuery = useSynthsBalancesQuery(walletAddress);
 	const isL2 = useRecoilValue(isL2State);
 	const { clearSubMenuConfiguration } = UIContainer.useContainer();
 
 	const snxBalance =
 		cryptoBalances?.balances?.find((balance) => balance.currencyKey === CryptoCurrency.SNX)
-			?.balance ?? toBigNumber(0);
+			?.balance ?? wei(0);
 
-	const sUSDBalance =
-		synthsBalancesQuery?.data?.balancesMap[Synths.sUSD]?.balance ?? toBigNumber(0);
+	const sUSDBalance = synthsBalancesQuery?.data?.balancesMap[Synths.sUSD]?.balance ?? wei(0);
 
 	const snxPriceChartData = useMemo(() => {
 		return (SNX24hrPricesQuery?.data ?? [])
-			.map((dataPoint) => ({ value: dataPoint.averagePrice }))
+			.map((dataPoint: { averagePrice: any }) => ({ value: dataPoint.averagePrice }))
```

Comment:
same as comment before here, i think we can derive the type for averagePrice from the graph query